Content/Items/Permafrost/Accessories.BookOfFrost.cs

```diff
@@ -0,0 +1,187 @@
+            }
+        }
+
+        private void CritExplosion(Player player, Item Item, NPC target, int damage, float knockback, bool crit)
+        {
+            if (Item.CountsAsClass(DamageClass.Melee) && crit)
+            {
+                Helper.PlayPitched("Magic/FrostHit", 0.75f, Main.rand.NextFloat(-0.05f, 0.05f), target.Center);
+                Projectile.NewProjectile(player.GetSource_OnHit(target), target.Center, Vector2.Zero, ModContent.ProjectileType<FrostExplosion>(), (int)(damage * 1.25f), knockback * 0.25f, player.whoAmI);
+            }
+        }
+
+        public override void SafeSetDefaults()
+        {
+            Item.rare = ItemRarityID.Green;
+            Item.value = Item.sellPrice(gold: 1, silver: 25);
+        }
+    }
+
+    class FrostExplosion : ModProjectile, IDrawPrimitive
+    {
+        private List<Vector2> cache;
+        private Trail trail;
+        private Trail trail2;
+
+        public float TimeFade => 1 - Projectile.timeLeft / 20f;
+        public float Radius => EaseBuilder.EaseCubicOut.Ease(1 - (Projectile.timeLeft / 20f)) * 55f;
+
+        public override string Texture => AssetDirectory.Invisible;
+
+        public override void SetDefaults()
+        {
+            Projectile.friendly = true;
+            Projectile.width = 120;
+            Projectile.height = 120;
+            Projectile.tileCollide = false;
+            Projectile.timeLeft = 20;
+            Projectile.penetrate = -1;
+            Projectile.usesLocalNPCImmunity = true;
+            Projectile.localNPCHitCooldown = 10;
+        }
+
+        public override void AI()
+        {
+            if (Main.netMode != NetmodeID.Server)
+            {
+                ManageCaches();
+                ManageTrail();
+            }
+
+            for (int k = 0; k < 8; k++)
+            {
+                float rot = Main.rand.NextFloat(0, 6.28f);
+
+                if (Main.netMode != NetmodeID.Server)
+                    Dust.NewDustPerfect(Projectile.Center + Vector2.One.RotatedBy(rot) * Radius, ModContent.DustType<Dusts.Glow>(), Vector2.One.RotatedBy(rot + Main.rand.NextFloat(1.1f, 1.3f)) * 2, 0,
+                        Main.rand.NextBool() ? new Color(100, 150 + (int)(100 * (float)Math.Sin(TimeFade * 3.14f)), 255) : new Color(0, 165 + (int)(100 * (float)Math.Sin(TimeFade * 3.14f)), 225), 0.4f);
+
+                for (int i = 0; i < 2; i++)
+                {
+                    var pos = Projectile.Center + Vector2.One.RotatedBy(rot) * Radius;
+                    var vel = Vector2.One.RotatedBy(rot + Main.rand.NextFloat(1.1f, 1.3f)) * 2;
+                    Dust.NewDustPerfect(pos, ModContent.DustType<Dusts.Gas>(), vel, 0, new Color(255, 255, 255) * 0.1f, Main.rand.NextFloat(4.5f, 6.5f));
+
+                    var pos2 = Projectile.Center + Vector2.One.RotatedBy(rot) * Radius;
```

Comment:
these appear to be identical in value to the other pos/vel local variables defined here -- why not simply use those?